api/src/main/java/jakarta/data/page/impl/PageRecord.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2003, 2013, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jakarta.data.page.impl;
+
+import jakarta.data.page.Page;
+import jakarta.data.page.PageRequest;
+import jakarta.data.page.Slice;
+
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Record type implementing {@link Slice}.
+ * This may be used to simplify implementation of a repository interface.
+ *
+ * @param pageRequest The {@link PageRequest page request} for which this
+ *                    slice was obtained
+ * @param content The page content
+ * @param totalElements The total number of elements across all pages that
+ *                      can be requested for the query
+ * @param <T> The type of elements on the page
+ */
+public record PageRecord<T>(PageRequest<T> pageRequest, List<T> content, long totalElements)
+        implements Page<T> {
+
+    @Override
+    public boolean hasContent() {
+        return !content.isEmpty();
+    }
+
+    @Override
+    public int numberOfElements() {
+        return content.size();
+    }
+
+    @Override @SuppressWarnings("unchecked")
+    public <E> PageRequest<E> pageRequest(Class<E> entityClass) {
+        return (PageRequest<E>) pageRequest;
+    }
+
+    @Override
+    public PageRequest<T> nextPageRequest() {
+        return pageRequest.next();
+    }
+
+    @Override @SuppressWarnings("unchecked")
+    public <E> PageRequest<E> nextPageRequest(Class<E> entityClass) {
+        return (PageRequest<E>) pageRequest.next();
+    }
+
+    @Override
+    public Iterator<T> iterator() {
+        return content.iterator();
+    }
+
+    @Override
+    public long totalPages() {
+        return totalElements / pageRequest.size();
```

Comment:
With 9 results and size 10, this will return 0 total pages, which is wrong.
With 10 results and size 10, this will return 1 total page, which is correct.
With 11 results and size 10, this will return 1 total page, which is wrong.

I believe something such as the following would correct it,
```suggestion
        int size = pageRequest.size();
        return totalElements / size + (totalElements % size > 0 ? 1 : 0);
```

Reply:
Yeah when I wrote this I didn't realize that we were using 1-based indexing. I wrote it assuming 0-based indexing.

Reply:
Although here I guess it's just a matter of using ceiling instead of floor. That's just a mistake.

Reply:
[See the updated commit.]